File: day1/solution_part2.py
#%%
import sys
sys.path.append('../')
import advent_of_code

input_list = advent_of_code.open_file("input.txt")

def count_increase(input_list):
    nums = [int(l) for l in input_list]
    counter = 0
    prev = []
    for i in range(len(nums)-2):
        curr = [nums[i],nums[i+1],nums[i+2]]
        if not prev:
            prev = curr
        elif sum(curr) > sum(prev):
            counter += 1
        prev = curr
    return counter

count_increase(input_list)

File: day1/solution_part2.c
#include <stdio.h>
#include <limits.h>

void main(){
    FILE *fp;
    int counter = 0;
    int i = 0;
    int list[5000];
    fp = fopen("input.txt","r");
    fscanf(fp,"%d",&i);
    while (!feof (fp))
    {
        fscanf (fp, "%d", &i);   
        list[counter] = i;
        counter++;
    }
    int higher = 0;
    int prev = INT_MAX;
    for (int j= 0;j<counter-2;j++){
        int curr = list[j]+list[j+1]+list[j+2];
        if (curr>prev){
            higher++;
        }
        prev = curr;
    }
    printf("%d\n",higher);

}